compiler/optimizer/RegisterCandidate.cpp

```diff
@@ -3847,24 +3828,28 @@ TR_RegisterCandidates::computeAvailableRegisters(TR_RegisterCandidate *rc, int32
          int32_t blockNumber = bvi.getNextElement();
          TR::Block * b = blocks[blockNumber];
 
-         if (!overlapTable->Exists(blockNumber))
+         if (overlapTable->find(blockNumber) == overlapTable->end())
             {
-            Coordinates c(comp()->allocator());
-            overlapTable->AddEntryAtPosition(blockNumber, c);
-            Coordinates &overlaps = (*overlapTable)[blockNumber];
-            ComputeOverlaps(b, comp(), overlaps);
+            Coordinates *c = new (comp()->trMemory()->currentStackRegion()) Coordinates((CoordinatesComparator()), (CoordinatesAllocator(comp()->trMemory()->currentStackRegion())));
```

Comment:
So there was a dance because of the copy construction - conversion to a pointer makes this particular bit easier to handle. Further, the Coordinate won't escape - the table is private to the opt, there is no getter, and the table is lexically allocated in the assign function and so the table dies once that scope ends. Stack allocation is , therefore, reasonable for this I believe.